runtime/src/bank.rs

```diff
@@ -3085,6 +3087,104 @@ impl Bank {
         }
     }
 
+    // TODO - solution 1, check vote redundancy by VoteState within bank,
+    //     this solution does not alight with bankless leader goal;
+    //     mainly using it to compare with solution #2 to confirm
+    //     behavior and performance.
+    fn is_redundant_vote_by_vote_state(&self, vote_account_pubkey: &Pubkey, vote: &Vote) -> bool {
```

Comment:
Would be good to add a test for this function

Reply:
Yes! Once we settled on final solution, Will add test for it if this is the chosen one

Reply:
thanks @carllin for helping setup test.